### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by tfrommen**
This replaces #16.

### CODE REVIEW

File: src/DistFile.php [hunk 1]
```diff
@@ -47,23 +47,19 @@ class DistFile
...
-                . $this->name
-            );
+        $this->fullpath = rtrim($fullpath, DIRECTORY_SEPARATOR);
+        $this->relativepath = rtrim($relativepath, DIRECTORY_SEPARATOR);
+        if ($this->name && $this->fullpath) {
+            $this->lastChangeTime = filemtime($this->fullpath . DIRECTORY_SEPARATOR . $this->name);
         }
-        $this->basepath = rtrim(str_replace($this->relativepath, null, $this->fullpath), DIRECTORY_SEPARATOR);
-        $this->distpathname = rtrim($this->relativepath, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . $this->name;
+        $this->basepath = preg_replace("~{$this->relativepath}$~", '', $this->fullpath);
```
Comment (reisraff) [inline src/DistFile.php]: Here the `rtrim` is used to remove the last DIRECTORY_SEPARATOR, do you have sure that in this case will not have the last DIRECTORY_SEPARATOR?

File: src/DistFile.php [hunk 2]
```diff
@@ -47,23 +47,19 @@ class DistFile
      */
     public function __construct(
         $content,
-        $name = null,
-        $fullpath = null,
-        $relativepath = null
+        $name = '',
```
Comment (reisraff) [inline src/DistFile.php]: why `''` instead `null`?

File: src/DistFile.php (hunk 1 above)
Comment (tfrommen) [inline src/DistFile.php]: I moved it to `$this->fullpath` and `$this->relativepath` above. Also, `Symfony/Finder` doesn't give paths with trailing DS.

Comment (reisraff) [inline src/DistFile.php]: Okay, but as you can see here https://github.com/reisraff/phulp-minifier/blob/master/src/JsMinifier.php#L48, the DistFile can be constructed by anyone, and we ***MUST*** make sure that the last DS will be removed.

**Pull request merged**